ChibiOS/PybStick2x/nanoBooter/main.c

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+#include <ch.h>
+#include <hal.h>
+#include <hal_nf_community.h>
+#include <cmsis_os.h>
+
+#include <usbcfg.h>
+#include <swo.h>
+#include <targetHAL.h>
+#include <WireProtocol_ReceiverThread.h>
+#include <nanoPAL_BlockStorage.h>
+#include <LaunchCLR.h>
+
+// need to declare the Receiver thread here
+osThreadDef(ReceiverThread, osPriorityHigh, 2048, "ReceiverThread");
+
+//  Application entry point.
+int main(void) {
+
+  // HAL initialization, this also initializes the configured device drivers
+  // and performs the board-specific initializations.
+  halInit();
+
+  // init SWO as soon as possible to make it available to output ASAP
+  #if (SWO_OUTPUT == TRUE)  
+  SwoInit();
+  #endif
+
+  // The kernel is initialized but not started yet, this means that
+  // main() is executing with absolute priority but interrupts are already enabled.
+  osKernelInitialize();
+  osDelay(20);    // Let init stabilize
+
+  // check for valid CLR image 
```

Comment:
Missing here the button check routine to remain on booter. It can be quite handy when you have a bad C# app deployed.